src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -79,30 +76,28 @@ describe.skip('ServerMetricsCollector', () => {
 
   it('collect disconnects requests infos', async () => {
     const never = new Promise((resolve) => undefined);
-    const hitSubject = new BehaviorSubject(0);
+    const disconnectRequested$ = new Subject(); // Controls the number of requests in the /disconnect endpoint
+    const disconnectAborted$ = new Subject(); // Controls the abort event in the /disconnect endpoint
 
     router.get({ path: '/', validate: false }, async (ctx, req, res) => {
       return res.ok({ body: '' });
     });
     router.get({ path: '/disconnect', validate: false }, async (ctx, req, res) => {
-      hitSubject.next(hitSubject.value + 1);
```

Comment:
I went for a plain `Subject` vs. `BehaviourSubject(counter)` just in case it was affected by concurrent read/writes:
1. The 2 `/disconnect` requests occur at a similar time
2. This line is executed when `hitSubject.value` still holds `0` for both requests.

IMO, for the purpose of the test, it's enough relying on the `.pipe(take(2))` approach. What do you think?

Reply:
`hitSubject.next(hitSubject.value + 1);` is synchronous, so 'atomical' for the event loop. I don't think there's any risk for concurrency problems. But using a plain subject and counting the emissions works fine too.

Reply:
Thanks for explaining! I was mostly concerned if calling `.next` would synchronously update `.value` to consider it atomic. Since it's a RxJS internal implementation, I was not sure. Glad to know that's the case!

Reply:
> I was mostly concerned if calling .next would synchronously update .value to consider it atomic

AFAIK it does (it evens call the subscribers synchronously), but I agree that we should probably consider this as an implementation detail of RXJS